backend/app/graphql/all_users.py

```diff
@@ -0,0 +1,25 @@
+import graphene
+from ..graphql.services import services
+from .types import QueryList, Query
+
+
+class User(Query):
+    first_name = graphene.String()
+    last_name = graphene.String()
+    email = graphene.String()
+    role = graphene.String()
+
+
+class UserQueries(QueryList):
+    users = graphene.List(
+        User,
+        first=graphene.Int(default_value=5),
+        offset=graphene.Int(default_value=0),
+        role=graphene.String(),
+    )
+
+    def resolve_users(self, info, first, offset, role=None):
+        user_service = services["user_service"]
+        users = user_service.get_users(role)
```

Comment:
Since the user_service iterates over all users right now, the filter below doesn't really add any efficiency. Instead, could we move the filtering inside the service as well (i.e. pass first and offset as additional arguments)? I'll add a comment to the service to say what I mean.